### PR TITLE
Issues/851 extend solution version and platfrom

### LINKED ISSUE #851: [Nox.Solution] Create PlatformId for Integration Event usage
related to #799 

# PlatformId And Version

For message identification a uri prefix will be used and configured in the Nox Solution:

1. Create a Property called PlatformId in the Solution root (next to the name)
    1. Required, but if not defined  default to solution name (so the user may not set it explicitly)
    2. Can not be set to null or empty 
    3. Its a string type
 1. Create a Property called Version (Version type) in the Solution root (next to the name)  
     2. Required, but if not defined  default 1.0 (so the user may not set it explicitly)  
     4.  Can not be set to null or empty

### PR DESCRIPTION
**Pull request by AndreyDegtyarev**
Introduced PlatformId and Version property in Solution class.

### CODE REVIEW

File: src/Nox.Solution/Models/Solution.cs [hunk 1]
```diff
@@ -10,11 +11,23 @@ namespace Nox.Solution;
...
-    public string Name { get; internal set; } = null!;
+    [Pattern(RegexConstants.SolutionNamePattern)]
+    public string Name { get; set; } = null!;
+
+    [Title("The message identification Platform Id.")]
+    [Description("For message identification a PlatformId  will be used and configured in the Nox Solution. Required, but if not defined default to solution name (so the user may not set it explicitly).")]
+    public string PlatformId { get => _platformId ?? Name; internal set => _platformId = value; }
+
+    [Required]
+    [Title("The version of the NOX solution.")]
```
Comment (rochar) [inline src/Nox.Solution/Models/Solution.cs]:
```suggestion
    [Title("The version of the NOX solution. Expected a Semantic Version format.")]
```

File: src/Nox.Solution/Models/Solution.cs [hunk 2]
```diff
@@ -10,11 +11,23 @@ namespace Nox.Solution;
...
+    [Description("Required, but if not defined default 1.0 (so the user may not set it explicitly).")]
```
Comment (rochar) [inline src/Nox.Solution/Models/Solution.cs]:
```suggestion
    [Description("Required, but if not defined defaults to 1.0 .")]
```

File: src/Nox.Solution/Models/Solution.cs [hunk 3]
```diff
@@ -10,11 +11,23 @@ namespace Nox.Solution;
...
+    [Pattern(RegexConstants.SolutionVersionPattern)]
+    public string Version { get; internal set; } = null!;
```
Comment (rochar) [inline src/Nox.Solution/Models/Solution.cs]: The default value is not being set?

File: src/Nox.Solution/Models/Solution.cs [hunk 4]
```diff
@@ -10,11 +11,23 @@ namespace Nox.Solution;
...
+
     [Required]
     [Title("The short name for the solution. Contains no spaces.")]
     [Description("The name of the NOX solution, application or service. This value is used extensively by the NOX tooling and libraries and should ideally be unique within an organisation.")]
-    [Pattern(@"^[^\s]*$")]
-    public string Name { get; internal set; } = null!;
+    [Pattern(RegexConstants.SolutionNamePattern)]
+    public string Name { get; set; } = null!;
+
+    [Title("The message identification Platform Id.")]
```
Comment (rochar) [inline src/Nox.Solution/Models/Solution.cs]:
```suggestion
    [Title("Platform Identifier. Used to build a unique Uri")]
```

File: src/Nox.Solution/Models/Solution.cs [hunk 5]
```diff
@@ -10,11 +11,23 @@ namespace Nox.Solution;
...
+    [Description("For message identification a PlatformId  will be used and configured in the Nox Solution. Required, but if not defined default to solution name (so the user may not set it explicitly).")]
```
Comment (rochar) [inline src/Nox.Solution/Models/Solution.cs]:
```suggestion
    [Description("Identify a Platform, that is a set of different services. Use to produce a unique Uri, by encoding the provided value.")]
```

File: samples/.nox/design/cryptocash.solution.nox.yaml
Comment (rochar) [inline samples/.nox/design/cryptocash.solution.nox.yaml:1]: no need to change all yamls, you need to have a default value and provide a test that asserts if th version is empty in the yaml, the value 1.0 is returned

**Review by rochar — CHANGES_REQUESTED** [review]

**Review by rochar — APPROVED** [review]
